Formula/gap.rb

```diff
@@ -11,46 +19,141 @@ class Gap < Formula
+      "toolsforhomalg", "toricvarieties",
+    ]
+
+    # These two packages either don't build or don't work
+    # "cddinterface", "xgap",
+    configure_packages = [
+      "anupq", "caratinterface", "crypting", "curlinterface", "cvec",
+      "datastructures", "deepthought", "digraphs", "ferret", "float",
+      "gauss", "grape", "io", "json", "normalizinterface", "nq",
+      "orb", "profiling", "semigroups", "simpcomp", "zeromqinterface",
+      ]
+
+    old_configure_packages = [
+      "ace", "browse", "cohomolo", "edim", "example", "fplsa",
+      "guava", "kbmag",
+    ]
+
+    # These package have autogen.sh available
+    # I don't think there is a need to run it, but we could if we wanted
+    autogen_packages = [
+      "anupq", "cddinterface", "curlinterface", "digraphs", "ferret",
+      "float", "guava", "io", "normalizinterface", "nq", "semigroups",
+      "simpcomp", "xgap", "zeromqinterface",
+    ]
+
+    # Run special commands after installation
+    special_packages = {
+      # Even with x11 installed, it doesn't seem to work
+      # "xgap" => "cp bin/xgap.sh $GAPROOT/bin/xgap.sh",
+    }
+
+
+    # Start actually building GAP
+    if build.head?
+      system "./autogen.sh"
+    end
+
+    libexec.install Dir["pkg"]
+
+    system "./configure", "--prefix", libexec/"gap", "--with-readline=#{Formula["readline"].opt_prefix}"
+    if build.head?
+      system "make", "bootstrap-pkg-full"
+      # system "make", "doc"      # Do we need this?
     end
+    system "make", "install"
 
     # Create a symlink `bin/gap` from the `gap` binary
-    bin.install_symlink libexec/"gap" => "gap"
+    bin.install_symlink libexec/"gap/bin/gap" => "gap"
 
     ohai "Building included packages. Please be patient, it may take a while"
+    pkg_dir = "#{libexec}/pkg"
+
     cd libexec/"pkg" do
-      # NOTE: This script will build most of the packages that require
-      # compilation. It is known to produce a number of warnings and
-      # error messages, possibly failing to build several packages.
-      system "../bin/BuildPackages.sh", "--with-gaproot=#{libexec}"
+
+      system "mkdir", "#{libexec}/gap/lib/gap/pkg/"
+
+      # The makefiles appear to only be used for docs...
+      # The BuildPackages.sh script didn't call them
+      no_compilation_packages.concat(makefile_packages).each do |pkg|
+        system "cp", "-R", pkg, "#{libexec}/gap/lib/gap/pkg/"
+      end
+
+      prerequisites_packages.each do |pkg|
+        cd pkg do
+          system "./prerequisites.sh", "#{libexec}/gap/lib/gap"
+        end
+      end
+
+      # autogen_packages.each do |pkg|
+      #   cd pkg do
+      #     system "./autogen.sh"
+      #   end
+      # end
+      configure_packages.each do |pkg|
+        cd pkg do
+          system "./configure", "--with-gaproot=#{libexec}/gap/lib/gap"
+          system "make"
+          system "/usr/bin/rsync", "-aEvL", "bin/", "#{libexec}/gap/lib/gap/pkg/"
```

Comment:
Why `rsync` ? Why not `cp -R` (or `cp -pR`) *all* packages? Then you don't need the `no_compilation_packages` or `makefile_packages` at all?

Reply:
When I was working on it, it kept causing problems.  I have changed a few things since then so I'll try `cp -R` again to see how it does.

Reply:
It still causes problems.  Several packages each have the directory `bin/x86_64-apple-darwin21-default64-kv8/` and it seems to want to copy that directory over as a whole rather than recursively copying the files in it.  `rsync` seems to do the right thing.

Reply:
You are *merging* those `bin` dirs into one. That's wrong! Each must go into the corresponding package directory

Reply:
Aha!  It felt like something was wrong.  Sorry for being a little dense.